upup/pkg/fi/cloudup/awstasks/subnet.go

```diff
@@ -261,6 +266,36 @@ func (_ *Subnet) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *Subnet) error {
 		}
 	}
 
+	if changes.DNS64 != nil {
+		request := &ec2.ModifySubnetAttributeInput{
+			SubnetId:    e.ID,
+			EnableDns64: &ec2.AttributeBooleanValue{Value: e.DNS64},
+		}
+		_, err := t.Cloud.EC2().ModifySubnetAttribute(request)
+		if err != nil {
+			return fmt.Errorf("error enabling DNS64 on subnet: %v", err)
+		}
+	}
+
+	if changes.ResourceBasedNaming != nil {
+		request := &ec2.ModifySubnetAttributeInput{
+			SubnetId:                                e.ID,
+			EnableResourceNameDnsAAAARecordOnLaunch: &ec2.AttributeBooleanValue{Value: e.ResourceBasedNaming},
```

Comment:
I think this should be keyed off of whether there are any IPv6 CIDRs assigned, not the value of `ResourceBasedNaming`.